server.go

```diff
@@ -372,8 +372,8 @@ func (s *Server) serveFile(w http.ResponseWriter, r *http.Request) {
 func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	// Run a cleanup in a goroutine only if not cleaning up at the moment
 	// and last cleanup was more than CleanupIntervalSeconds seconds ago.
-	if atomic.LoadInt64(&s.fileCleanerLastRun) != 0 {
-		if time.Now().Unix() - atomic.LoadInt64(&s.fileCleanerLastRun) > int64(s.Conf.CleanupIntervalSecs) {
+	if time.Now().Unix() - atomic.LoadInt64(&s.fileCleanerLastRun) > int64(s.Conf.CleanupIntervalSecs) {
+		if atomic.CompareAndSwapInt64(&s.fileCleanerLastRun, 0, time.Now().Unix()) {
```

Comment:
I don't think it does what the comment says. 

CompareAndSwapT is the atomic equivalent of the following code:
```
if *addr == old {
	*addr = new
	return true
}
return false
```

Since 0 indicates already running cleanup. It reads "write time.Now() only when cleanup is already running and then start cleanup".

`sync/atomic` is hard. Perhaps, just use a mutex?

Reply:
You see, when last run is 0, it calculates as greater than CleanupIntervalSecs and executes the swap as the next check passes, too.

Reply:
It is unsafe to use mutex if database transactions will block for some reason.

Reply:
Looks like you just reimplemented sync.Mutex using channels anyway, isn't it?
You can keep mutex locked for a short period of time while you check whether you need to run cleanupFiles or not and release it after.

E.g.
```go
lock()
if should_cleanup && !cleanup_running {
  go cleanup()
}
unlock()
//proceed as normal
```

cleanup:
```go
lock()
cleanupRunning = true
unlock()
actualCleanup()
```